doc/arm_jogging/arm_jogging_tutorial.rst

```diff
@@ -0,0 +1,126 @@
+
+    rosservice call /controller_manager/switch_controller "start_controllers:
+    - 'joint_group_position_controller'
+    stop_controllers:
+    - 'arm_controller'
+    strictness: 2"
+
+
+Launch the jog node. This example uses commands from a `SpaceNavigator <https://www.google.com/search?client=ubuntu&channel=fs&q=amazon+buy+spacenavigator&ie=utf-8&oe=utf-8>`_ joystick-like device: ::
+
+    roslaunch jog_arm spacenav_cpp.launch
+
+If you do not have a SpaceNav 3D mouse, you can publish example jogging commands from the command line with: ::
+
+    rostopic pub -r 100 /jog_arm_server/delta_jog_cmds geometry_msgs/TwistStamped "header: auto
+    twist:
+      linear:
+        x: 0.0
+        y: 0.01
+        z: -0.01
+      angular:
+        x: 0.0
+        y: 0.0
+        z: 0.0"
+
+Settings
+--------
+User-configurable settings of the jog node are well-documented in ``jog_arm/config/ur_simulated_config.yaml``.
+
+Robot Requirements
+------------------
+The jogger streams an array of position or velocity commands to the robot controller. This is compatible with ros\_control ``position_controllers/JointGroupPositionControllers`` or ``velocity_controllers/JointGroupVelocityControllers``. You can check if these controllers are available for your robot by searching for the controller config file (typically named ``controllers.yaml``). After launching the robot, you can check if any ros_control controllers are available with: ::
+
+    rosservice call /controller_manager/list_controllers
+
+And switch to the desired controller with: ::
+
+    rosservice call /controller_manager/switch_controllers controller_to_start controller_to_stop
+
+**NOTE:** You can tab-complete to help fill these commands.
+
+Jogging may work on other robots that have a different control scheme but there is no guarantee. It has been tested heavily on UR robots using the [ur_modern_driver](https://github.com/ros-industrial/ur_modern_driver). The jogger currently does not limit joint jerk so may not be compatible with most heavy industrial robots.
+
+The jogger can publish ``trajectory_msgs/JointTrajectory`` or ``std_msgs/Float64MultiArray`` message types. This is configured in a yaml file (see ``config/ur_simulated_config.yaml`` for an example).
+
+ROS Signals
+-----------
+An `rqt_graph` of the jogger is shown below (Enlarge by clicking it). Most of these connections can be ignored. The important ones are:
+
+- **spacenav_to_twist** node: Converts incoming commands from the joystick to Cartesian commands or joint angle commands, depending on which buttons are pressed.
+
+- **tf**: This topic carries ROS coordinate frame information. The jogger uses it to transform commands from the joystick's frame of reference to the robot's frame of reference. These frames are selected in ``config/ur_simulated_config.yaml``.
```

Comment:
Which frames are the important ones in the robot frame? Only the base frame or any other robot frames as well?
If other frames are used as well, isn't the delay of TF causing issues there?

Reply:
Yes, the delay of tf might cause issues in fast applications. Not for the typical teleop scenario, though

Reply:
You didn't yet answer my first question: Which frames are the important ones in the robot frame?
As you also get the `joint_states`, I guess you compute both FK and IK in your server, so only the base frame of the robot is required from TF. Would be great to confirm this in the README.

Reply:
Added a comment that everything is done in the `world` frame.

Reply:
world is a specific frame name that is not defined in all robots. I guess you want to refer to the robot's planning / base frame.